### PR TITLE
feat: add basic magic sync

### PR DESCRIPTION
**Pull request by Darkwood9612**
> [!IMPORTANT]
> Add new Magic API for spell casting and animation management, with supporting services and message types.
> 
>   - **Magic API**:
>     - Add `MagicApi` with functions `castSpellImmediate`, `interruptCast`, `getAnimationVariablesFromActor`, and `applyAnimationVariablesToActor`.
>     - Handles spell casting and animation variables.
>   - **Services**:
>     - Add `MagicSyncService` in `magicSyncService.ts` for synchronizing magic events.
>     - Update `NetworkingService` to handle new message types `SpellCast` and `UpdateAnimVariables`.
>   - **Messages**:
>     - Add `SpellCastMessage` and `UpdateAnimVariablesMessage` in `spellCastMessage.ts` and `updateAnimVariablesMessage.ts`.
>     - Extend `MsgType` enum with `SpellCast` and `UpdateAnimVariables`.
>   - **Event Handling**:
>     - Improve `TESSpellCastEvent` handling to ensure `spellCast` event is always triggered.
>   - **Documentation**:
>     - Add `sp-magic-api.md` to document new magic API functions.
>   - **Misc**:
>     - Added `EquippedItemType.SpellOrScroll` to `EquippedItemType` enum.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=skyrim-multiplayer%2Fskymp&utm_source=github&utm_medium=referral)<sup> for 4e385df69d67d03fb0c7d1b0491f29e43b687508. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp [hunk 1]
```diff
@@ -1326,44 +1328,63 @@ EventResult EventHandler::ProcessEvent(
       return;
     }
 
-    auto castingSource = RE::MagicSystem::CastingSource::kLeftHand;
-    bool isCasterValid = false;
+    const bool isLeftHand =
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:1331]: changelog needed https://github.com/skyrim-multiplayer/skymp/tree/main/docs/release/dev

**Review by Pospelove — CHANGES_REQUESTED** [review]
actionlistener.cpp: spdlog::info(fmt::format) -> spdlog::info

actionlistener.cpp: "SentToNeighbors" -> "RealFunctionName - " in logging

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp [hunk 2]
```diff
@@ -1326,44 +1328,63 @@ EventResult EventHandler::ProcessEvent(
...
 
     AddObjProperty(&obj, "caster", caster, "ObjectReference");
     AddObjProperty(&obj, "target", handleTarget, "ObjectReference");
     AddObjProperty(&obj, "spell", spell, "Spell");
-    AddObjProperty(&obj, "isDualCasting", magicCaster->GetIsDualCasting());
+    AddObjProperty(&obj, "isDualCasting", isDualCasting);
     AddObjProperty(&obj, "castingSource",
                    static_cast<uint32_t>(castingSource));
 
+    auto animVariables =
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:1373]: plz move anim vars out of the event

**Comment by Pospelove** [comment]
TODO:
- Reverse-engineer and sync flames/bolts direction

File: skyrim-platform/src/platform_se/codegen/convert-files/skyrimPlatform.ts [hunk 3]
```diff
@@ -1601,10 +1602,18 @@ export interface Inventory {
   entries: Entry[];
 }
 
-export declare function setInventory(formId: number, inventory: Inventory): void;
+export interface ActorAnimationVariables {
+  Booleans: ArrayBuffer
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/codegen/convert-files/skyrimPlatform.ts:1606]: Booleans => booleans
Floats => floats
Integers => integers

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp (hunk 2 above)
Comment (Darkwood9612) [reply skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp]: done

File: skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp (hunk 1 above)
Comment (Darkwood9612) [reply skyrim-platform/src/platform_se/skyrim_platform/EventHandler.cpp:1358]: done

File: skyrim-platform/src/platform_se/codegen/convert-files/skyrimPlatform.ts (hunk 3 above)
Comment (Darkwood9612) [reply skyrim-platform/src/platform_se/codegen/convert-files/skyrimPlatform.ts:1606]: done

**Comment by Pospelove** [comment]
this is a good start. merging. we will resolve TODOs in follow ups

**Pull request merged**